pkg/liquidity-mining/contracts/gauges/avalanche/AvalancheRootGauge.sol

```diff
@@ -70,66 +126,109 @@ contract AvalancheRootGauge is StakelessGauge {
      * @dev Must be deployed by the AvalancheRootGaugeFactory, or other contract that implements
      * `IAvalancheBridgeLimitsProvider`.
      */
-    constructor(IMainnetBalancerMinter minter, IMultichainV4Router multichainRouter) StakelessGauge(minter) {
-        _minter = minter;
-        _multichainRouter = multichainRouter;
-        _bridgeLimitsProvider = IAvalancheBridgeLimitsProvider(msg.sender);
+    constructor(IMainnetBalancerMinter minter, ILayerZeroBALProxy lzBALProxy) StakelessGauge(minter) {
+        _lzBALProxy = lzBALProxy;
+        uint8 decimalDifference = ERC20(address(minter.getBalancerToken())).decimals() - lzBALProxy.sharedDecimals();
```

Comment:
BAL will always be 18 decimals on mainnet (and probably everywhere else foreseeable). Maybe we could hard-code that and remove some dependencies?

Reply:
We still need the minter anyways for the underlying stakeless gauge. So this doesn't really add an extra dependency.